heroku/deploy-a-django-app.md

```diff
@@ -0,0 +1,272 @@
+plugin](https://devcenter.heroku.com/changelog-items/1441).
+
+### Create Heroku config files
+
+Define config files relative to the root of your repo.
+
+#### `heroku.yml`
+
+The `heroku.yml` manifest file tells Heroku how to build and network the services
+and containers that comprise your application. This file should live in the root of your project repo.
+For background and syntax, see [the
+documentation](https://devcenter.heroku.com/articles/build-docker-images-heroku-yml).
+Use the following baseline to get started:
+
+```yml
+# Define addons that you need for  your project, such as Postgres, Redis, or Solr.
+setup:
+  addons:
+    - plan: heroku-postgresql
+# Define your application's Docker containers.
+build:
+  docker:
+    web: Dockerfile
+# Define any scripts that you'd like to run every time the app deploys.
+release:
+  command:
+    - ./scripts/release.sh
+  image: web
+# The command that runs your application. Replace 'app' with the name of your app.
+run:
+  web: gunicorn -t 180 --log-level debug app.wsgi:application
+```
+
+#### `release.sh`
+
+In your app's `scripts` folder, define a script `release.sh` to run every time the app deploys.
+Use the following baseline script:
+
+```bash
+#!/bin/bash
+python manage.py collectstatic --noinput
+python manage.py migrate --noinput
+```
+
+If your app uses a `dbload` script to load initial data into the database, you can use `release.sh`
+to check if the initial data exists and run the data loading scripts if not. For example:
+
+```bash
+# Set ${TABLE} to the name of a table that you expect to have data in it.
+if [ `psql ${DATABASE_URL} -tAX -c "SELECT COUNT(*) FROM ${TABLE}"` -eq "0" ]; then
+    make all
+fi
+```
+
+#### `app.json`
+
+In order to enable [review apps](https://devcenter.heroku.com/articles/github-integration-review-apps)
+for your project, the repo must contain an `app.json` config file in the root directory.
+For background and syntax, see [the documentation](https://devcenter.heroku.com/articles/app-json-schema).
+Use the following baseline to get started:
+
+```json
+{
+  "name": "your-app",
+  "description": "Short description of your app.",
+  "scripts": {},
+  "env": {
+    "DJANGO_SECRET_KEY": {
+      "required": true
+    }
+  },
+  "formation": {
+    "web": {
+      "quantity": 1,
+      "size": "hobby"
+    }
+  },
+  "addons": [
+    "heroku-postgresql"
+  ],
+  "buildpacks": [],
+  "stack": "container"
+}
+```
+
+### Create apps and pipelines for your project
+
+In order to deploy your project, you need to create Heroku apps for staging
+and production, and tie them together in a [pipeline](https://devcenter.heroku.com/articles/pipelines).
+To create these resources, start by defining the name of your app:
+
+```bash
+# This should be the same slug as your project's GitHub repo.
+export APP_NAME=<your-app-name-here>
+```
+
+Then, run the following Heroku CLI commands:
+
+```bash
```

Comment:
I would have found it useful to have a version of these instructions where I could create the staging application, but not the production application, especially when the application has expensive add-ons (e.g., Solr).